lib/Localization/LocalizationFormat.cpp

```diff
@@ -96,6 +96,10 @@ bool SerializedLocalizationWriter::emit(llvm::StringRef filePath) {
 llvm::StringRef
 LocalizationProducer::getMessageOr(swift::DiagID id,
                                    llvm::StringRef defaultMessage) {
+  if (state == NotInitialized) {
+    initialize();
+    state = Initialized;
+  }
```

Comment:
That way we will try to initialize the localization producer only once. Right?

Reply:
See my comment for `initialize()` declaration.

Reply:
Also this should check for `state == FailedInitialization` and produce default if so.